src/Graphs/AdjacencyListDirectedGraph.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Bearded.Utilities.Graphs
+{
+    class AdjacencyListDirectedGraph<T> : IDirectedGraph<T> where T : IEquatable<T>
+    {
+        private readonly ImmutableList<T> elements;
+        private readonly ImmutableDictionary<T, ImmutableList<T>> directSuccessors;
+        private readonly ImmutableDictionary<T, ImmutableList<T>> directPredecessors;
+
+        public IEnumerable<T> Elements => elements;
+        public int Count => elements.Count;
+
+        public AdjacencyListDirectedGraph(
```

Comment:
Consider making this internal so nobody can sub-class this outside the binary?

Reply:
There classes themselves are already internal, but done anyway in case that changes in the future.